### PR TITLE
Improved Repository Setup

### PR DESCRIPTION
**Pull request by ivanzvonkov**
**What changed:**
- Wrote additional instructions about setting up environment and processing data
- Added environment yml with strict package versions
- Added Tensorboard for logging loss 
- Cleaned up dvg.py so that it requires minimal changes on setup

**Why**
- This PR makes it easier to set up the repository and train a model with KTH data

### CODE REVIEW

File: DVG.py [hunk 1]
```diff
@@ -1,19 +1,19 @@
+import warnings
+warnings.simplefilter("ignore", UserWarning)
```
Comment (ivanzvonkov) [inline DVG.py:2]: gpytorch uses a legacy autograd function and pytorch throws too many warnings about this.

File: DVG.py [hunk 2]
```diff
@@ -207,7 +223,7 @@ def train(x,epoch):
     optimizer.step()
 
 
-    return mse_latent.data.cpu().numpy()/(opt.n_past+opt.n_future),mse_latent.data.cpu().numpy()/(opt.n_past+opt.n_future)
+    return mse_latent.data.cpu().numpy()/(opt.n_past+opt.n_future)
```
Comment (ivanzvonkov) [inline DVG.py:226]: Redundant

File: DVG.py [hunk 3]
```diff
@@ -313,24 +331,26 @@ def plot(x, epoch):
...
             x,y = next(training_batch_generator)
-            mse_ctrl, indices = train(x,epoch) 
+            mse_ctrl = train(x,epoch) 
             epoch_mse += mse_ctrl 
 
         progress.finish()
         utils.clear_progressbar()
 
-        print('[%02d] mse loss: %.5f (%d) %.5f' % (epoch, epoch_mse/opt.epoch_size, epoch*opt.epoch_size*opt.batch_size, indices))
+        print('[%02d] mse loss: %.5f (%d) %.5f' % (epoch, epoch_mse/opt.epoch_size, epoch*opt.epoch_size*opt.batch_size, mse_ctrl))
```
Comment (ivanzvonkov) [inline DVG.py:340]: indices and mse_ctrl contained same value

File: data/meta_creator.py [hunk 4]
```diff
@@ -1,23 +1,36 @@
...
+    data = {"train": [], "test": []}
+    for file in files:
+        if 'train' in file or 'test' in file:
+            continue
+
+        temp = os.listdir(start_dir + clas+'/'+file)
+        temp.sort()
+        dic = {
+            'vid': file,
+            'files': [i.tolist() for i in np.array_split(temp, 4)],
```
Comment (ivanzvonkov) [inline data/meta_creator.py:22]: Since I am saving as json everything has to be a list

File: data/meta_creator.py [hunk 5]
```diff
@@ -1,23 +1,36 @@
...
-			temp = os.listdir(start_dir + clas+'/'+file)
-			temp.sort()
-			dic['files'] = np.array_split(temp,4)
-			dic['n'] = len(temp)
-			lists.append(dic)
-	torch.save(lists,'%s/%s/train_meta64x64.pt'%(start_dir,clas))
+    files = os.listdir(start_dir + clas)
+    files.sort()
+    train = [f"person{i:02}_" for i in range(0, 17)]
+    test = [f"person{i:02}_" for i in range(17, 26)]
```
Comment (ivanzvonkov) [inline data/meta_creator.py:12]: `kth.py` has the following code:
```python
if train:
    self.train = True
    data_type = 'train'
    self.persons = list(range(1, 21))
```
However, I don't see `.persons` actually being used anywhere. So I decided to make the train/test split in here, hence the above two lines.

File: data/meta_creator.py [hunk 6]
```diff
@@ -1,23 +1,36 @@
 import os
-import torch
 import numpy as np
+import json
 
-start_dir = 'processed/'
+start_dir = 'data/processed/'
```
Comment (ivanzvonkov) [inline data/meta_creator.py:5]: This is the one file that needs a serious look as some logic is different.

File: environment.yml [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+  - pytorch=1.2.0
+  - scikit-learn=0.23.2
+  - scipy=1.1.0
+  - torchvision=0.4.0
+  - pip:
+    - future==0.18.2
+    - gpytorch==0.3.0
+    - ipython==7.24.1
+    - progressbar==2.5
+    - tensorboard==2.5.0
```
Comment (ivanzvonkov) [inline environment.yml:19]: @the-darklord could you compare the versions of cudatoolkit, pytorch, gpytorch to what you have in your environment and let me know if they differ?

**Comment by ivanzvonkov** [comment]
@shgaurav1 did you have a chance to look at this?
